pkg/proxy/apis/proxyconfig/register.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package proxyconfig
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var (
+	SchemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
+	AddToScheme   = SchemeBuilder.AddToScheme
+)
+
+// GroupName is the group name use in this package
+const GroupName = "kubeproxy.k8s.io"
```

Comment:
There is still some debate as to whether this should be called kubeproxy.k8s.io or kubeproxyconfig.k8s.io. Copying the comments from #53833.

I wrote

> There was also a question in the kube-proxy PR about naming the group kubeproxyconfig.k8s.io vs kubeproxy.k8s.io. It sounds somewhat redundant to have apiVersion=kubeproxyconfig.k8s.io/v1beta1 with kind=KubeProxyConfiguration. What do others think?

@mtaufen wrote

> That's an interesting point. If you intend the group to only ever contain the kube-proxy config objects, I'd still call it kubeproxyconfig.k8s.io to make that clear.


We need a final decision.

@smarterclayton @jbeda @deads2k @liggitt @sttts @mtaufen @mikedanese @luxas

Reply:
The kubelet is doing kubeletconfig instead of just kubelet, so maybe we should go with kubeproxyconfig.k8s.io, despite the redundancy of group=kubeproxyconfig.k8s.io, Kind=KubeProxyConfig?

Reply:
I think that is okay. This group is all about configuration anyway; you might possibly put other config-related things there as well.

Reply:
grouping by component makes sense to me. putting `config` in the group name feels stuttery. `KubeProxyConfig` in `kubeproxy.k8s.io/v1beta1` seems pretty well-qualified.

Reply:
that said, if you have a component that takes config, and serves an API (like the kubelet), I can see wanting to separate out the config API objects and the served API objects

Reply:
Ok, @xiangpengzhao please:

1. Rename the api group to `kubeproxyconfig.k8s.io`
1. Rename the directory `pkg/proxy/apis/proxyconfig` to `kubeproxyconfig`